### ISSUE TITLE
Update MongoDB driver to 3.0

### ISSUE THREAD

**Issue by ChrisDeBruyne** [lordofthejars/nosql-unit#133]
We can't run our tests anymore since some methodes have been removed.
Like : 

java.lang.NoSuchMethodError: com.mongodb.DB.isAuthenticated()Z
        at
com.lordofthejars.nosqlunit.mongodb.MongoOperation.getMongoDb(MongoOperation.java:122)

**Comment by lordofthejars** [comment]
Yes This would be a big impact on code upgrading to mongo driver 3.0. Currently I am working to implementing Neo4j2.x

I will need to see what's happening with MongoDb3.0. Of course feel free to collaborate.

**Comment by ChrisDeBruyne** [comment]
Would like to, but your master branch is unstable now. What commit is best
to branch from?

**Comment by lordofthejars** [comment]
Well it is unstable because of neo4j, I will try to fix this weekend, but
you can work with mongoDB module without any problem and just run these
tests.

El dv., 8 maig 2015 a les 17:26, ChrisDeBruyne (<notifications@github.com>)
va escriure:

> Would like to, but your master branch is unstable now. What commit is best
> to branch from?

**Comment by ChrisDeBruyne** [comment]
I was checking out the tests and noticed that you use Fongo (https://github.com/fakemongo/fongo)
This is a problem because Fongo also does not support the new Java driver.
So unless you decide to drop Fongo and use the real deal in your unit tests, I'm stuck :-(

**Comment by lordofthejars** [comment]
Everything done. Next release will contain MongoDB support.

**Comment by paulovictorv** [comment]
I'm really sorry to revive an old issue, but is there any updates on this? I've noticed that the only problem seems to be Fongo. 

I've added an exclusion to ``1.5.10`` version of Fongo, and manually added the latest ``2.0.2`` and everything is running smooth.

**Comment by ricardozanini** [comment]
In fact, there's a problem with `DBRef` instances too:
`java.lang.NoSuchMethodError: com.mongodb.DBRef.getRef()Ljava/lang/String;`

I am getting this error within `@UsingDataSet` and my data files pointing to each other through `$ref` attributes. I saw that this issue is already addressed in the new version, though. Waiting for this new release as well. :)

Thanks for your time and work with this lib, lord. But I just want to get this registered for others looking for this problem too. Downgrade to Mongo 2 is impossible for me at this time.

**Comment by jschutijzer** [comment]
Do you have any idea when the next release will be? We need the upgrade urgently :)

**Comment by lordofthejars** [comment]
Now tests are integrated with mongo 3.3.X, I will release this night.

**Comment by lordofthejars** [comment]
Released 0.12.0, now waiting to be sync to maven central.
